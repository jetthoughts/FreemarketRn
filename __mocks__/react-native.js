const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
