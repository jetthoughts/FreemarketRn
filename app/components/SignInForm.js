import React, { Component } from 'react';
import {
  View,
  Text,
  TextInput,
  Button,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  base: {
    flex: 1,
    padding: 40,
  },
  input: {
    height: 40,
  },
  button: {
    paddingTop: 10,
  },
});

export default class SignInForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      login: '',
      password: '',
    };
  }

  submit() {
    const { login, password } = this.state;
    this.props.onSubmit({ login, password });
  }

  render() {
    return (
      <View style={styles.base}>
        <Text>Login:</Text>
        <TextInput
          style={styles.input}
          onChangeText={login => this.setState({ login })}
          value={this.state.login}
        />
        <Text>Password:</Text>
        <TextInput
          style={styles.input}
          onChangeText={password => this.setState({ password })}
          value={this.state.password}
          secureTextEntry
        />
        <View style={styles.button}>
          <Button
            title="Log In"
            onPress={this.submit}
          />
        </View>
      </View>
    );
  }
}

SignInForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};
