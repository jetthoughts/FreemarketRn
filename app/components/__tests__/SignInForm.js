import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import SignInForm from '../SignInForm';

it('renders successfully', () => {
  const tree = renderer.create(
    <SignInForm
      onSubmit={() => true}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
