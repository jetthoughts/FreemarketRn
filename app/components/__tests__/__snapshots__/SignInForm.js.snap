// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 40,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Login:
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "height": 40,
      }
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Password:
  </Text>
  <TextInput
    onChangeText={[Function]}
    secureTextEntry={true}
    style={
      Object {
        "height": 40,
      }
    }
    value=""
  />
  <View
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Log In
        </Text>
      </View>
    </View>
  </View>
</View>
`;
